python/pyspark/mllib/linalg.py

```diff
@@ -633,6 +743,57 @@ def stringify(vector):
         """
         return str(vector)
 
+    @staticmethod
+    def dot(a, b):
+        """
+        Dot product between two vectors.
+        a and b can be of type, SparseVector, DenseVector, np.ndarray
+        or array.array.
+
+        >>> a = Vectors.sparse(4, [(0, 1), (3, 4)])
+        >>> b = Vectors.dense([23, 41, 9, 1])
+        >>> Vectors.dot(a, b)
+        27.0
+        >>> Vectors.dot(a, a)
+        17.0
+        >>> Vectors.dot(a, np.array([0, 1, 2, 4]))
+        16.0
+        """
+        a, b = _convert_to_vector(a), _convert_to_vector(b)
+        return a.dot(b)
+
+    @staticmethod
+    def squared_distance(a, b):
```

Comment:
There's an inconsistency here:
* Python: ```DenseVector.squared_distance```, ```SparseVector.squared_distance```
* Scala: ```Vectors.sqdist```

I'm OK with naming this squared_distance or sqdist, or with having both with one as an alias.  CC @mengxr  Votes?